### ISSUE TITLE
CMakeLists: Unsetting CMAKE_REQUIRED_FLAGS incorrect?

### ISSUE THREAD

**Issue by GhostGambler** [chgagne/beagle#6]
The following line is likely intended to undo the changes introduced to check for PACC: `set(CMAKE_REQUIRED_FLAGS)		# Unset modified CMake variables`
https://github.com/chgagne/beagle/blob/2eadc45b316bad2e41052e7129a45b7bb51f2a7a/CMakeLists.txt#L443

I expect this to set CMAKE_REQUIRED_FLAGS to an empty string? Then this might break further tests, because CMAKE_REQUIRED_FLAGS might contain other values as well: `set(CMAKE_REQUIRED_FLAGS "-std=c++0x")`
https://github.com/chgagne/beagle/blob/2eadc45b316bad2e41052e7129a45b7bb51f2a7a/CMakeLists.txt#L209

The old value should for example be saved in a temp variable and written back in line 443

**Comment by mgard** [comment]
You are right. Actually, there are many wrong practices like that in the CMakeList. It was mainly made about 6 years ago by an undergraduate intern, with an old version of CMake.

We will review the whole CMakeList (there are many cleanups we can do, including removing old options that are not useful anymore, like the check for GCC 2.95...), and, in the process, take your suggestions into account. I leave this issue open as a reminder.

Thanks for your report.
